src/main.py

```
@@ -780,6 +784,99 @@ async def analyze_research_trends(
+        logger.info("SSE event generator cancelled.")
+        # Handle cleanup if necessary
+    except Exception as e:
+        logger.error(f"Error in SSE event generator: {e}")
+        # Yield an error event if possible, or just log and exit
+        try:
+            yield {
+                "event": "error",
+                "data": json.dumps({"error": str(e)})
+            }
+        except: # If yielding fails (e.g. connection already closed)
+            pass
```

Ruff (0.11.9)

Use `contextlib.suppress(Exception)` instead of `try`-`except`-`pass`

(SIM105)